rotkehlchen/assets/resolver.py

```diff
@@ -111,6 +114,36 @@ def get_asset_type(identifier: str, query_packaged_db: bool = True) -> AssetType
         instance.types_cache.add(identifier, asset_type)
         return asset_type
 
+    @staticmethod
+    def check_existence(identifier: str, query_packaged_db: bool = True) -> str:
+        """Check that an asset with the given identifier exists and return normalized identifier
+
+        For example if 'eTh' is given here then 'ETH' should be returned.
+
+        May raise:
+        - UnknownAsset: If asset identifier does not exist.
+        """
+        # TODO: This is ugly here but is here to avoid a cyclic import in the Assets file
+        # Couldn't find a reorg that solves this cyclic import
+        from rotkehlchen.constants.assets import CONSTANT_ASSETS  # pylint: disable=import-outside-toplevel  # isort:skip  # noqa: E501
+        from rotkehlchen.globaldb.handler import GlobalDBHandler   # pylint: disable=import-outside-toplevel  # isort:skip  # noqa: E501
```

Comment:
We should look into this. I'm not sure if there is a performance hit due to having the import here

Reply:
yes. And it's not just this function. Many functions in this part. We really need to solve this dependency.